### PR TITLE
Modified footer to allow for responsive links. Modified row usage to …

### PR DESCRIPTION
**Pull request by trevorsaint**
closes: https://github.com/ONSdigital/design-system/issues/1127

- Simplified macro
- Split row usage and added `legal` to simplify use
- List links are now fluid and don’t require breakpoint classes
- Documentation added

This will potentially be a breaking change. Instead of using two rows, we will need to add legal to include things like:

- Cookies
- Accessibility statement
- Privacy and data protection
- Terms and conditions

We can still add many rows into the footer but this spits our legal links out, which allows more layout control.

**Before:**
Screenshot 2020-11-09 at 16 47 11

**After:**
Screenshot 2020-11-09 at 16 46 47

### CODE REVIEW

**Comment by rmccar** [comment]
[Issue #1127: Footer should use a fluid layout, allowing links to wrap in a row](https://trello.com/c/k6JGrwhf/756-issue-1127-footer-should-use-a-fluid-layout-allowing-links-to-wrap-in-a-row)

File: src/scss/utilities/_grid.scss [hunk 1]
```diff
@@ -121,7 +125,7 @@
 .grid__col {
   display: inline-block;
   box-sizing: border-box;
-  flex: 0 0 auto;
+  // flex: 0 0 auto;
```
Comment (rmccar) [inline src/scss/utilities/_grid.scss:128]: Probably just want to remove this rather than comment it out

**Comment by rmccar** [comment]
image

Theres an issue with the footer warning

File: src/scss/utilities/_grid.scss (hunk 1 above)
Comment (trevorsaint) [reply src/scss/utilities/_grid.scss:128]: Agreed.

**Review by rmccar — APPROVED** [review]

**Pull request merged**